automation-packages/playwright-automation-packages-with-library/README.md

```diff
@@ -0,0 +1,214 @@
+# Playwright Automation Packages with Library
+
+This project demonstrates how to structure a multi-module Maven project for Step automation packages using a shared library approach. It illustrates best practices for creating reusable, versioned code and sharing resources (like Playwright drivers) across multiple automation packages.
+
+## Overview
+
+This project consists of four modules that work together to demonstrate a scalable approach to automation package development:
+
+### 1. playwright-automation-package-library
+
+The **Library** module serves as the foundation for all automation packages in this project. It:
+
+- Provides an `AbstractPlaywrightKeyword` base class that extends Step's `AbstractKeyword`
+- Implements **driver sharing** across keywords from different Automation Packages through Step's session management
+- Ensures proper resource lifecycle management with automatic cleanup
+- Is built as an **uber-jar** (fat jar) packaging all dependencies including Playwright
+- Can be versioned independently and reused across multiple projects
+
+**Key Feature**: The library demonstrates how to share a Playwright driver instance between keywords, even when those keywords are deployed from different Automation Packages. This is achieved using Step's session storage mechanism (`session.put()` / `session.get()`), which ensures the same browser instance is reused across keyword executions within the same session.
+
+### 2. playwright-automation-package-opencart
+
+The **OpenCart** module is a concrete Automation Package that:
+
+- Depends on the shared library module
+- Contains Keywords declared in Java code for testing the OpenCart demo application
+- Extends `AbstractPlaywrightKeyword` to leverage shared driver functionality
+- **NOT** building as an uber-jar (regular jar packaging)
+
+Example keyword: `"OpenCart - Buy MacBook in OpenCart"`
+
+### 3. playwright-automation-package-exense-website
```

Comment:
We could have used the Webmail Keyword from the playwright E2E sample (can be done later)